src/test/java/org/folio/rest/impl/APITests.java

```diff
@@ -1,99 +1,178 @@
-  protected static final String OKAPI_TOKEN = buildToken();
-  protected static final String OKAPI_HEADER_URL = "x-okapi-url";
+  private static Vertx vertx;
 
-  protected static Vertx vertx;
-  protected static OkapiClient okapiClient;
-  protected static TenantClient tenantClient;
+  protected final ResourceClient accountsClient = accountsClient();
 
-  @BeforeClass
-  public static void beforeAll(final TestContext context) throws Exception {
-    Async async = context.async();
+  @ClassRule
+  public static WireMockRule wireMock = new WireMockRule(
+    new WireMockConfiguration().dynamicPort());
+
+  protected final OkapiClient client = new OkapiClient(getOkapiUrl());
 
+  @BeforeClass
+  public static void deployVerticle() throws Exception {
     vertx = Vertx.vertx();
-    okapiClient = new OkapiClient(OKAPI_URL, OKAPI_TENANT, OKAPI_TOKEN);
-    tenantClient = new TenantClient(OKAPI_URL, OKAPI_TENANT, OKAPI_TOKEN);
 
     PostgresClient.getInstance(vertx).startEmbeddedPostgres();
 
-    DeploymentOptions deploymentOptions = new DeploymentOptions()
-      .setConfig(new JsonObject().put(HTTP_PORT_PROPERTY, OKAPI_PORT));
+    final CompletableFuture<Void> future = new CompletableFuture<>();
 
-    vertx.deployVerticle(RestVerticle.class.getName(), deploymentOptions, deployment -> {
-      try {
-        tenantClient.postTenant(getTenantAttributes(), result -> async.complete());
-      } catch (Exception e) {
-        log.error(e.getMessage());
-        context.fail(e);
-      }
-    });
+    vertx.deployVerticle(RestVerticle.class.getName(), createDeploymentOptions(),
+      res -> future.complete(null));
+
+    future.get(10, TimeUnit.SECONDS);
+
+    mockEndpoints();
+    createTenant();
   }
 
   @AfterClass
-  public static void afterAll(final TestContext context) {
-    Async async = context.async();
-    vertx.close(context.asyncAssertSuccess(res -> {
+  public static void undeployEnvironment() throws Exception {
+    final CompletableFuture<Void> future = new CompletableFuture<>();
+
+    vertx.close(notUsed -> {
       PostgresClient.stopEmbeddedPostgres();
-      async.complete();
-    }));
+      future.complete(null);
+    });
+
+    future.get(10, TimeUnit.SECONDS);
+  }
+
+  @Before
+  public void restoreMocks() {
+    mockEndpoints();
+  }
+
+  private static void mockEndpoints() {
+    wireMock.resetAll();
+
+    wireMock.stubFor(post(urlMatching("/pubsub/publish"))
+      .atPriority(100)
+      .willReturn(noContent()));
+    wireMock.stubFor(post(urlMatching("/pubsub/event-types/declare/(publisher|subscriber)"))
+      .atPriority(100)
+      .willReturn(created()));
+    wireMock.stubFor(post(urlEqualTo("/pubsub/event-types"))
+      .atPriority(100)
+      .willReturn(created()));
+
+    // forward everything to okapi
+    wireMock.stubFor(any(anyUrl())
+      .atPriority(Integer.MAX_VALUE)
+      .willReturn(aResponse().proxiedFrom("http://localhost:" + OKAPI_PORT)));
+  }
```

Comment:
Could this be extracted to a class of it's own?

Reply:
Moved to OkapiDeployment class.